### PR TITLE
Fixes order by id button in library's computer

### LINKED ISSUE #16180: can't order books with SBN code
### What is the problem?

Type book code in the library computer
### Why is it a problem, or what should have happened?

nothing happens

### LINKED ISSUE #17176: Library Archive Issues (ss13bn/title search)
**BUG:** It seems that ordering books by the ss13bn (book number) is just completely broken.  It doesn't matter what number you put in, it's broken and does not order the book at all. 

**FEEDBACK/USABILITY ISSUE:** For this reason, an alternative solution would be to ctrl-F the title of the book but since the archive was changed to only display a certain number of books per page to prevent crashing or something, it will only search those books on the page, so you have to flip page and page and ctrl-F every single page to finally find your book.  Maybe a search bar or some function like that could be implemented into the archive interface?

### LINKED ISSUE #21253: library archive system is busted
For a good year or so, the library archive system has been nothing but trouble.  Originally, all the books uploaded into the external archive showed up on the librarian's computer in one long list.  Sometime last year, this list was segregated into pages of book titles.  Ever since this happened, the game would crash whenever you try to go past page 3.  Trying to order books by their SSBN doesn't work, either.  Trying to read some books would also crash the game in a similar manner, although I don't know why this happens.
This is what the crashing error looks like:
http://i.imgur.com/vOy663K.png

What I would like is if someone could import /vg/ station's library system.  It's much more stable, and allows for several search options when you're trying to print out books.  I don't know how easy it would be to do this, but I would vastly appreciate it.

http://i.imgur.com/QvDGuCT.png

### PR DESCRIPTION
**Pull request by lzimann**
:cl: Lzimann
fix: You can once again order books by its ID in the library.
/:cl:

The ID needed to be passed as a text so sanitizeSQL would work.

Fixes #21253 
~~Couldn't reproduce the crash after a few pages.~~ the crashing was something else and has been fixed.

Fixes #17176
Fixes #16180

### CODE REVIEW

**Comment by Jordie0608** [comment]
Library code is a pain but the things that should be changed are conveniently just outside the file diff here so ¯\\\_(ツ)_/¯

**Comment by bgobandit** [comment]
where is the Paid Code tag

File: code/modules/library/lib_machines.dm [hunk 1]
```diff
@@ -410,10 +410,14 @@ var/global/list/datum/cachedbook/cachedbooks // List of our cached book datums
...
-		if(orderid)
-			if(isnum(orderid) && IsInteger(orderid))
-				href_list["targetid"] = orderid
+		if(bibledelay)
+			say("Printer unavailable. Please allow a short time before attempting to print.")
+		else
+			var/orderid = input("Enter your order:") as num|null
+			if(orderid)
+				if(isnum(orderid) && IsInteger(orderid))
+					href_list["targetid"] = num2text(orderid) // This allows orderid to be sanitized.
```
Comment (RemieRichards) [inline code/modules/library/lib_machines.dm]: the comment is unnecessary.

File: code/modules/library/lib_machines.dm [hunk 2]
```diff
@@ -432,17 +436,17 @@ var/global/list/datum/cachedbook/cachedbooks // List of our cached book datums
...
 				B.icon_state = "book[rand(1,8)]"
-				src.visible_message("[src]'s printer hums as it produces a completely bound book. How did it do that?")
+				visible_message("[src]'s printer hums as it produces a completely bound book. How did it do that?")
 				break
-	src.add_fingerprint(usr)
-	src.updateUsrDialog()
-	return
+
+	add_fingerprint(usr)
+	//updateUsrDialog()
```
Comment (RemieRichards) [inline code/modules/library/lib_machines.dm]: Also this is what updates the page, so be very sure you're removing it for good reason (ie: duplicates)

**Review by RemieRichards — APPROVED** [review]

File: code/modules/library/lib_machines.dm (hunk 2 above)
Comment (lzimann) [inline code/modules/library/lib_machines.dm]: It was already like that, I commented on accident(testing something) and forgot to uncomment.

**Pull request merged**